### PR TITLE
copr: add more assert in compare_in test

### PR DESCRIPTION
**Pull request by TennyZhuang**
###  What have you changed?

Related to #6000 optimization, a simple and hidden bug #6093 is introduced in the last round of review and broke the optimization, but the test still pass.

The bug is fixed in #6093, and this PR improve the test to ensure the optimization is effective.

###  What is the type of the changes?

Pick one of the following and delete the others:

- Improvement (a change which is an improvement to an existing feature)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test

### CODE REVIEW

**Comment by TennyZhuang** [comment]
@breeswish @sticnarf PTAL

File: components/tidb_query/src/rpn_expr/impl_compare_in.rs [hunk 1]
```diff
@@ -235,10 +235,10 @@ mod tests {
     use crate::codec::batch::{LazyBatchColumn, LazyBatchColumnVec};
     use crate::expr::EvalContext;
     use crate::rpn_expr::types::RpnFnMeta;
-    use crate::rpn_expr::RpnExpressionBuilder;
+    use crate::rpn_expr::{RpnExpressionBuilder, RpnExpressionNode};
 
     #[test]
-    fn test_in() {
+    fn test_in_constant() {
```
Comment (breeswish) [inline components/tidb_query/src/rpn_expr/impl_compare_in.rs]: Prefer to add a test that include both constants and non-constants.

Comment (TennyZhuang) [reply components/tidb_query/src/rpn_expr/impl_compare_in.rs]: `test_in_complex` include both constants and non-constants, I think that a `test_in_constant` is more obvious to show the result.

Comment (breeswish) [reply components/tidb_query/src/rpn_expr/impl_compare_in.rs]: Then how about also check remaining constants in test_in_complex?

Comment (TennyZhuang) [reply components/tidb_query/src/rpn_expr/impl_compare_in.rs]: done

**Pull request merged**